### PR TITLE
(chore): bump dependencies to resolve CVE issues

### LINKED ISSUE #6143: CVE-2022-23648 CVE-2022-31030 github.com/containerd/containerd package
#### What did you do?
I have executed a vulnerability scan on the latest available image: 

- quay.io/operator-framework/helm-operator:latest

#### What did you expect to see?
A vulnerability free image

#### What did you see instead? Under which circumstances?
Two vulnerabilities on the github.com/containerd/containerd package: 

1. CVE-2022-23648
2. CVE-2022-31030

#### Environment

**Operator type:** Helm Operator

**Kubernetes cluster type:** Not applicable

`$ operator-sdk version`

- Not applicable

`$ go version` (if language is Go)

`$ kubectl version`

- Not applicable

#### Possible Solution

Upgrade the following package to the latest available version:

- github.com/containerd/containerd

### LINKED ISSUE #6322: Please update go-restful golang-jwt and golang.org/x/net version to fix twistlock scan issue
Hello,
ansible-operator get below security issues in twistlock scan. Please update versoin in go.mod accordingly . Thank you.

high | H | 7.5 | Has fix,High severity | PRISMA-2022-0227 | https://github.com/emicklei/go-restful/issues/497 | Y | fixed in v3.10.0 | go | github.com/emicklei/go-restful/v3 | v3.9.0 |   |   | /usr/local/bin/ansible-operator
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
medium | M | 5.4 | DoS,Has fix,Medium severity | PRISMA-2022-0270 | https://github.com/golang-jwt/jwt/issues/223 | Y | fixed in v4.4.3 | go | github.com/golang-jwt/jwt/v4 | v4.2.0 |   |   | /usr/local/bin/ansible-operator
low | L | 1 | DoS,Has fix,Recent vulnerability | CVE-2022-41723 | https://nvd.nist.gov/vuln/detail/CVE-2022-41723 | Y | fixed in 0.7.0 | go | golang.org/x/net | v0.0.0-20221014081412-f15817d10f9b | /usr/local/bin/ansible-operator

it still old go-restful version:
https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L95

it still old golang-jwt:
https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L117

it still old golang.org/x/net:
https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L218

### LINKED ISSUE #6405: Please update github.com/docker/docker, github.com/containerd/containerd, helm.sh/helm/v3 version to fix twistlock scan issue
Hi,
helm-operator get below security issues in twistlock scan. Please update versoin in go.mod accordingly . Thank you.

 high | H | 7 | Has fix, High Severity | CVE-2023-28840 | github.com/docker/docker | Y | fixed in 23.0.3, 20.10.24 | go | current version github.com/docker/docker v20.10.19 
 -- | -- | -- | -- | -- | --| -- | -- | -- | -- 
 Meidum | M | 4 | Has fix, Medium Severity | CVE-2023-25153 | github.com/containerd/containerd | Y | fixed in 1.6.18, 1.5.18 | go | current version github.com/containerd/containerd v1.6.8 
 Meidum | M | 4 | Has fix, Medium Severity | CVE-2023-25165 | helm.sh/helm/v3 | Y | fixed in 3.11.1 | go | current version helm.sh/helm/v3 v3.10.3

#### What did you do?

still uses helm.sh/helm/v3  version in go.mod file:
https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L36

still uses github.com/docker/docker  version in go.mod file:
https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L82

still uses github.com/containerd/containerd in go.mod file:
https://github.com/operator-framework/operator-sdk/blob/master/go.mod#L72

### LINKED ISSUE #6450: upgrade mongo-go-driver for CVE-2021-20329
## Bug Report
CVE Advisory: https://github.com/advisories/GHSA-f6mq-5m25-4r72 

Operator-sdk is currently using `github.com/golang-migrate/migrate/v4 v4.6.2` ([source](https://github.com/operator-framework/operator-sdk/blob/master/go.mod#LL112C2-L112C57)) which uses `mongo-go-driver v1.1.0` ([source](https://github.com/golang-migrate/migrate/blob/v4.6.2/go.mod#L44)). All versions of mongo-go-driver that are <1.5.1 are affected by this CVE. Could it please be updated?

### PR DESCRIPTION
**Pull request by everettraven**
**Description of the change:**
- Bump the following dependencies to ensure we aren't using versions resulting in twistlock scan issue:
    - github.com/emicklei/go-restful/v3 --> v3.10.0
    - github.com/golang-jwt/jwt/v4 --> v4.5.0
    - golang.org/x/net --> v0.10.0
    - helm.sh/helm/v3 --> v3.11.3
    - github.com/containerd/containerd --> v1.7.0
    - github.com/docker/docker --> v20.10.24
- Bump the `github.com/operator-framework/operator-registry` dependency to `v1.28.0` to pull in some related cve fixes

**Motivation for the change:**
- fixes #6322 
- fixes #6450
- fixes #6405
- fixes #6143

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -109,6 +109,7 @@ require (
 	github.com/gobwas/glob v0.2.3 // indirect
 	github.com/gofrs/uuid v4.0.0+incompatible // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/golang-jwt/jwt/v4 v4.4.3 // indirect
```
Comment (everettraven) [inline go.mod:112]: I'm not sure we are actually able to force this. This is an indirect dependency as a result of pulling in `github.com/distribution/distribution/v3` - which looks like it is pulled in via `helm.sh/helm/v3`.

Unfortunately `github.com/distribution/distribution/v3` has yet to update their golang-jwt version to one with a fix

**Comment by ukumar009** [comment]
Hi @everettraven 

I see you are not able to merge this Pull Request due to `github.com/distribution/distribution/v3`

Could you please consider merging this CVE `github.com/emicklei/go-restful/v3 --> v3.10.0` in a separate PR? Because this is a High Vulnerability CVE.

Or May I create a Separate Pull Request for the `github.com/emicklei/go-restful/v3`? Apologies for the inconvenience.

Thanks,

File: go.mod (hunk 1 above)
Comment (everettraven) [reply go.mod]: This has been updated and pulled in with https://github.com/operator-framework/operator-sdk/pull/6383/commits/5a865e1aa13b885e5b782b840755e232c1796b09

**Comment by everettraven** [comment]
@ukumar009 Apologies for the delayed response! I just got some time to circle back around to this. I am planning on updating this PR to fix a number of CVE related issues.

**Review by rashmigottipati — APPROVED** [review]
/lgtm

**Pull request merged**